LevelSpaceXWGUI.nlogo

```diff
@@ -572,7 +589,7 @@ to-report make-variadic-task [astring args]
   set astring add-spaces astring
   foreach string:rex-split astring "\\s" [
     ifelse member? ? args[
-      set sb lput (word "(item " (position ? args) " ?)") sb
+      set sb lput (word "(item " (position ? args) " ?" arg-num ")") sb
```

Comment:
My thoughts exactly. I wonder how much syntax-string manipulation could be put into an extension, with the goal being that eventually that extension would just become a macro system.
I'm going to think out loud here for a minute:
```netlogo
let zipped-args map list args (map [word "?" x] [1 2 3 4 5 6 7 8 9])
syntax:replace astring zipped-args
```
`syntax:replace` could break down a string into an AST-esque shape and carry out syntactically safe replacements. Perhaps it could also be used for generating syntax:
```netlogo
syntax:task ["arg1" "arg2"] ["rt arg1 fd arg2"]
```

Just some thoughts.